src/git-town-rename-branch

```diff
@@ -3,30 +3,40 @@ source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
 
 
 function preconditions {
-  target_branch_name=$1
-  destination_branch_name=$2
-  force_parameter=$3
-
   force_rename=false
-  if [ "$force_parameter" == "-f" ]; then
-    force_rename=true
-  fi
 
-  if [ "$(is_main_branch "$target_branch_name")" = true ]; then
+  if (( $# == 0 )); then
     echo_error_header
-    echo_error "The main branch cannot be renamed."
+    echo_error "No branch name provided to rename to."
     exit_with_error newline
-  fi
-
-  if [ -z "$target_branch_name" ]; then
+  elif (( $# == 1 )); then
+    target_branch_name=$(get_current_branch_name)
+    destination_branch_name=$1
+  elif (( $# == 2 )); then
+    if [[ $2 == "-f" ]]; then
+      target_branch_name=$(get_current_branch_name)
+      destination_branch_name=$1
+      force_rename=true
+    else
+      target_branch_name=$1
+      destination_branch_name=$2
+    fi
+  elif (( $# == 3 )); then
+    if [[ $3 == "-f" ]]; then
+      force_rename=true
+    fi
```

Comment:
I would think having a third parameter that isn't "-f" should be an error. Should add a test for this if we don't have one already